src/hex_utils.c

```diff
@@ -78,3 +78,45 @@ uint64_t hex_string_to_num(const size_t max_digits, const char *const str)
 	}
 	return ret;
 }
+
+/* This function attempts to read a number, starting from the given input pointer and stores the
+ * result in val.
+ * It performs a number of helpful additional tasks needed for number parsing in BMD.
+ * It stores the location of the character after the last one considered into rest if non-NULL
+ * this is useful for chaining these calls or to parse mutliple fields.
+ * It accepts a char parameter called follow, which if not set to READ_HEX_NO_FOLLOW will check
+ * the character after the number is equal to that character and report failure if it is not.
+ * If the follow character is matched then the value stored into the rest pointer will be after
+ * this following character.
+ * This routine also accepts a base to operate on as it is the common function used by the two
+ * wrappers read_hex32 and read_dec32 for hexadecimal and decimal numbers respectively.
+ * 
+ * This routine uses strtoul internally so all the number parsing behaviour of that apply,
+ * it will accept a leading + or - and perform negation of the read number correctly.
+ * If passed 0 as the base it will accept hex numbers prefixed with 0x or 0X and octal numbers
+ * prefixed with 0, otherwise it will assume decimal.
+ * It will skip white space preceding the number to be read.
+ * 
+ * The resturn value indicates whether a number has been successfully read, if the function returns
+ * true then rest and val have been assigned if non-NULL.
+ */
+bool read_unum32(const char *input, const char **rest, uint32_t *val, char follow, int base)
+{
+	char *end = NULL;
+	uint32_t result = strtoul(input, &end, base);
+	if (end == NULL || end == input)
+		return false;
+	else {
```

Comment:
For me removing this else results in the function growing by 2 bytes and the total flash usage increasing by 8 bytes.

I will make the change anyway but this seems to be because a conditional store for one of the out parameters got moved out of line requiring a backward jump which adds code size.

Reply:
Ah, that's not ideal 😔 still, hopefully that's an optimiser issue we can work around fairly easily, should the need arise.